core/src/main/scala/org/http4s/otel4s/OtelMetrics.scala

```diff
@@ -0,0 +1,235 @@
+    new MetricsOps[F] {
+      override def increaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .inc(
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+          )
+
+      override def decreaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .dec(
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+          )
+
+      override def recordHeadersTime(
+          method: Method,
+          elapsed: Long,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.requestDuration
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedAttributes.httpRequestMethod(method))
+              .added(TypedMetricAttributes.httpPhase(Phase.Headers)),
+          )
+
+      override def recordTotalTime(
+          method: Method,
+          status: Status,
+          elapsed: Long,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.requestDuration
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedAttributes.httpRequestMethod(method))
+              .added(TypedAttributes.httpResponseStatusCode(status))
+              .added(TypedMetricAttributes.httpPhase(Phase.Body)),
+          )
+
+      override def recordAbnormalTermination(
+          elapsed: Long,
+          terminationType: TerminationType,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.abnormalTerminations
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedMetricAttributes.errorType(terminationType)),
+          )
+
+      private def secondsFromNanos(nanos: Long): Double =
+        nanos / 1000000000.0
+    }
+
+  private def createMetricsCollection[F[_]: Monad: Meter](
+      prefix: String,
+      responseDurationSecondsHistogramBuckets: BucketBoundaries,
+  ): F[MetricsCollection[F]] = {
+    val requestDuration: F[Histogram[F, Double]] =
+      Meter[F]
+        .histogram[Double](s"$prefix.request.duration")
+        .withUnit("s")
+        .withDescription("Duration of HTTP server requests.")
+        .withExplicitBucketBoundaries(responseDurationSecondsHistogramBuckets)
+        .create
+
+    val activeRequests: F[UpDownCounter[F, Long]] =
+      Meter[F]
+        .upDownCounter[Long](s"$prefix.active_requests")
+        .withUnit("{request}")
+        .withDescription("Number of active HTTP server requests.")
+        .create
+
+    val abnormalTerminations: F[Histogram[F, Double]] =
+      Meter[F]
+        .histogram[Double](s"$prefix.abnormal_terminations")
```

Comment:
out of curiosity, why is this `Double` and not `Long`? can `Histogram` not be of `Long`?

Reply:
Currenly I'm recording the elapsed seconds with nanosecond precision for "request.duration" and "abnormal_terminations". I find the metric name "abnormal_terminations" pretty unintuitive but I took that from the prometheus middleware. I should probably set the unit to "s".

Reply:
oh, I must have misread. I thought it was just a count. nvm